src/lib/services-manager/widgets/services_table.rb

```diff
@@ -0,0 +1,311 @@
+
+      # Title for start_mode column
+      #
+      # @return [String]
+      def start_mode_title
+        _('Start')
+      end
+
+      # Title for state column
+      #
+      # @return [String]
+      def state_title
+        _('State')
+      end
+
+      # Title for description column
+      #
+      # @return [String]
+      def description_title
+        _('Description')
+      end
+
+      # Id for a table row of a service
+      #
+      # @param service_name [String]
+      # @return [Yast::Term]
+      def row_id(service_name)
+        Id(service_name)
+      end
+
+      # Value for the name column of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def name_value(service_name)
+        max_width = max_column_width(:name)
+        return service_name if service_name.size < max_width
+
+        service_name[0..(max_width - 3)] + "..."
+      end
+
+      # Value for the start_mode column of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def start_mode_value(service_name)
+        ServicesManagerService.start_mode_to_human_for(service_name)
+      end
+
+      # Value for the state column of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def state_value(service_name)
+        state = TRANSLATIONS[:service_state][service_state(service_name)]
+        substate = TRANSLATIONS[:service_substate][service_substate(service_name)]
+
+        return _(state) unless substate
+
+        format(_("%{state} (%{substate})"), state: _(state), substate: _(substate))
+      end
+
+      # Value for the description column of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def description_value(service_name)
+        ServicesManagerService.description(service_name) || ""
+      end
+
+      # State of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def service_state(service_name)
+        ServicesManagerService.state(service_name) || ""
+      end
+
+      # Substate of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def service_substate(service_name)
+        ServicesManagerService.substate(service_name) || ""
+      end
+
+      # Updates the value for the start_mode column of a service
+      #
+      # @param service_name [String]
+      def refresh_start_mode_value(service_name)
+        UI.ChangeWidget(id, Cell(service_name, 1), start_mode_value(service_name))
+      end
+
+      # Updates the value for the state column of a service
+      #
+      # @param service_name [String]
+      def refresh_state_value(service_name)
+        is_active = ServicesManagerService.find(service_name).active?
+        will_be_active = ServicesManagerService.active?(service_name)
+
+        state = if is_active != will_be_active
+          will_be_active ? _('Active (will start)') : _('Inactive (will stop)')
```

Comment:
this does not make sense for me. It is probably not active and will start, not? so "Inactive (will start)" and "active (will stop)"? This way it really does not make sense for me.

Reply:
Well, actually this is something to discuss with Ken. I guess we should mark somehow the temporary values (state or start mode). For example, if user changes start mode from :on_boot to :on_demand, the table should indicate that the value :on_demand is not applied yet. One option is to add a * to the string. For example "Active *", "On Demand *".